### PR TITLE
[terra-navigation-side-menu] A11y changes

### PR DESCRIPTION
**Pull request by MadanKumarGovindaswamy**
### Summary

**What was changed:**

-- Added headerLevel prop for the title of the menu.
-- Added missing A11y attributes.

**Why it was changed:**

Part of A11y changes.

### Testing

This change was tested using:

- [ ] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [x] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [x] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-10270 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-navigation-side-menu/src/_MenuItem.jsx [hunk 1]
```diff
@@ -137,6 +137,14 @@ class MenuItem extends React.Component {
...
     );
 
+    const ariaAttributes = {};
+    ariaAttributes['aria-haspopup'] = hasChevron;
+    ariaAttributes['aria-disabled'] = isDisabled;
+
+    if (hasChevron) {
+      ariaAttributes['aria-expanded'] = !hasChevron;
+    }
+
```
Comment (saket2403) [inline packages/terra-navigation-side-menu/src/_MenuItem.jsx:147]: Why do we need if condition here? If it is to check undefined / null then we should have default values declared for all rops.

Comment (MadanKumarGovindaswamy) [reply packages/terra-navigation-side-menu/src/_MenuItem.jsx:147]: we don't want add aria-expanded for all the list items, so only when it has submenu then append this attribute. like if we don't add this condition then SR reads as collapsed for all list items even the item does not have submenu.

File: packages/terra-navigation-side-menu/translations/es-ES.json [hunk 2]
```diff
@@ -1,3 +1,5 @@
 {
-  "Terra.navigation.side.menu.selected": "Seleccionado"
+  "Terra.navigation.side.menu.selected": "Seleccionado",
+  "Terra.navigation.side.menu.folder": "Folder",
+  "Terra.navigation.side.menu.document": "Document"
```
Comment (saket2403) [inline packages/terra-navigation-side-menu/translations/es-ES.json:4]: I think you should log translations

File: packages/terra-navigation-side-menu/src/NavigationSideMenu.module.scss [hunk 3]
```diff
@@ -84,4 +84,15 @@
     width: 100%;
     word-wrap: break-word; /* For IE 10 and IE 11 */
   }
+
+  .drill-in-title {
```
Comment (sugan2416) [inline packages/terra-navigation-side-menu/src/NavigationSideMenu.module.scss:88]: Can we keep the common styles between title and drill-in-title in .title and conditionally add-in the differences in .drill-in-title

Comment (MadanKumarGovindaswamy) [reply packages/terra-navigation-side-menu/src/NavigationSideMenu.module.scss:91]: Updated. Thanks

File: packages/terra-navigation-side-menu/translations/es-ES.json (hunk 2 above)
Comment (MadanKumarGovindaswamy) [reply packages/terra-navigation-side-menu/translations/es-ES.json:4]: yeah we need to log, will check with UX team once on this. Thanks

**Review by saket2403 — APPROVED** [review]

**Review by sugan2416 — APPROVED** [review]

**Comment by rbsree** [comment]
+1, for accessibility review for Terra Navigation Side Menu - Drillin example.

**Pull request merged**